lib/collection/src/shards/shard.rs

```diff
@@ -63,9 +63,13 @@ impl Shard {
         }
     }
 
-    pub fn get_telemetry_data(&self, detail: TelemetryDetail) -> LocalShardTelemetry {
+    pub async fn get_telemetry_data(&self, detail: TelemetryDetail) -> LocalShardTelemetry {
         let mut telemetry = match self {
-            Shard::Local(local_shard) => local_shard.get_telemetry_data(detail),
+            Shard::Local(local_shard) => {
+                let mut shard_telemetry = local_shard.get_telemetry_data(detail);
+                shard_telemetry.status = Some(local_shard.local_shard_info().await.status);
+                shard_telemetry
+            }
```

Comment:
We only do it this way because we want to expose the status to cluster manager, am I right?

Reply:
Looks kinda ugly... 😬

I'd add a comment explaining why is it structured like this (e.g., why not just move `local_shard_info` inside `get_telemetry_data`).